ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/T8nSubCommand.java

```diff
@@ -0,0 +1,557 @@
+          config.set("env", t8nReader.readTree(reader));
+        }
+      }
+      if (!alloc.equals(stdinPath)) {
+        try (FileReader reader = new FileReader(alloc.toFile(), StandardCharsets.UTF_8)) {
+          config.set("alloc", t8nReader.readTree(reader));
+        }
+      }
+      if (!txs.equals(stdinPath)) {
+        try (FileReader reader = new FileReader(txs.toFile(), StandardCharsets.UTF_8)) {
+          config.set("txs", t8nReader.readTree(reader));
+        }
+      }
+
+      referenceTestEnv = objectMapper.convertValue(config.get("env"), ReferenceTestEnv.class);
+      initialWorldState =
+          objectMapper.convertValue(config.get("alloc"), ReferenceTestWorldState.class);
+      initialWorldState.persist(null);
+      Iterator<JsonNode> it = config.get("txs").elements();
+
+      transactions = extractTransactions(it);
+      if (!outDir.toString().isBlank()) {
+        outDir.toFile().mkdirs();
+      }
+    } catch (final JsonProcessingException jpe) {
+      parentCommand.out.println("File content error: " + jpe);
+      jpe.printStackTrace();
+      return;
+    } catch (final IOException e) {
+      LOG.error("Unable to read state file", e);
+      return;
+    }
+
+    Log4j2ConfiguratorUtil.setLevel(
+        "org.hyperledger.besu.ethereum.mainnet.AbstractProtocolScheduleBuilder", Level.OFF);
+    final ReferenceTestProtocolSchedules referenceTestProtocolSchedules =
+        ReferenceTestProtocolSchedules.create(
+            new StubGenesisConfigOptions().chainId(BigInteger.valueOf(chainId)));
+    Log4j2ConfiguratorUtil.setLevel(
+        "org.hyperledger.besu.ethereum.mainnet.AbstractProtocolScheduleBuilder", null);
+
+    final MutableWorldState worldState = new DefaultMutableWorldState(initialWorldState);
+
+    final HeaderBasedProtocolSchedule protocolSchedule =
+        referenceTestProtocolSchedules.getByName(fork);
+    if (protocolSchedule == null) {
+      throw new UnsupportedForkException(fork);
+    }
+
+    ProtocolSpec protocolSpec =
+        protocolSchedule.getByBlockHeader(BlockHeaderBuilder.createDefault().buildBlockHeader());
+    final BlockHeader blockHeader = referenceTestEnv.updateFromParentValues(protocolSpec);
+    final MainnetTransactionProcessor processor = protocolSpec.getTransactionProcessor();
+    final WorldUpdater worldStateUpdater = worldState.updater();
+    final ReferenceTestBlockchain blockchain = new ReferenceTestBlockchain(blockHeader.getNumber());
+
+    List<TransactionReceipt> receipts = new ArrayList<>();
+    List<RejectedTransaction> invalidTransactions = new ArrayList<>();
+    List<Transaction> validTransactions = new ArrayList<>();
+    ArrayNode receiptsArray = objectMapper.createArrayNode();
+    long gasUsed = 0;
+    for (int i = 0; i < transactions.size(); i++) {
+      Transaction transaction = transactions.get(i);
+
+      final Stopwatch timer = Stopwatch.createStarted();
+      final OperationTracer tracer; // You should have picked Mercy.
```

Comment:
?? 😆

Reply:
Internet joke.  https://www.youtube.com/watch?v=d0RmRJsgP28
It's also here - https://github.com/shemnon/besu/blob/19818121e73fa47a99ad72d8871e3ba824dcb62c/ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java#L311

t8n tool copied a lot of the execution framework from the main evm tool tracer.